pkg/operator/starter.go

```diff
@@ -3,136 +3,95 @@ package operator
 import (
 	"context"
 	"fmt"
-	"os"
-	"time"
 
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/apimachinery/pkg/fields"
-	"k8s.io/client-go/dynamic"
+	dynamicclient "k8s.io/client-go/dynamic"
+	kubeclient "k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
 	"k8s.io/klog"
 
-	apiclientset "github.com/openshift/client-go/config/clientset/versioned"
-	apiinformers "github.com/openshift/client-go/config/informers/externalversions"
-
 	"github.com/openshift/library-go/pkg/controller/controllercmd"
-	"github.com/openshift/library-go/pkg/operator/loglevel"
-	"github.com/openshift/library-go/pkg/operator/management"
+	"github.com/openshift/library-go/pkg/operator/csi/csicontrollerset"
+	goc "github.com/openshift/library-go/pkg/operator/genericoperatorclient"
+	"github.com/openshift/library-go/pkg/operator/v1helpers"
 
-	"github.com/openshift/aws-ebs-csi-driver-operator/pkg/common"
-	clientset "github.com/openshift/aws-ebs-csi-driver-operator/pkg/generated/clientset/versioned"
-	informers "github.com/openshift/aws-ebs-csi-driver-operator/pkg/generated/informers/externalversions"
+	"github.com/openshift/aws-ebs-csi-driver-operator/pkg/apis/operator/v1alpha1"
+	"github.com/openshift/aws-ebs-csi-driver-operator/pkg/generated"
 )
 
 const (
-	resync = 20 * time.Minute
+	// Operand and operator run in the same namespace
+	defaultNamespace = "openshift-cluster-csi-drivers"
+	operatorName     = "aws-ebs-csi-driver-operator"
+	operandName      = "aws-ebs-csi-driver"
 )
 
 func RunOperator(ctx context.Context, controllerConfig *controllercmd.ControllerContext) error {
-	ctrlClientset, err := clientset.NewForConfig(controllerConfig.KubeConfig)
+	// Create clientsets and informers
+	kubeClient := kubeclient.NewForConfigOrDie(rest.AddUserAgent(controllerConfig.KubeConfig, operatorName))
+	dynamicClient := dynamicclient.NewForConfigOrDie(rest.AddUserAgent(controllerConfig.KubeConfig, operatorName))
+	kubeInformersForNamespaces := v1helpers.NewKubeInformersForNamespaces(kubeClient, defaultNamespace, "")
+
+	// Create GenericOperatorclient. This is used by the library-go controllers created down below
+	gvr := v1alpha1.SchemeGroupVersion.WithResource("awsebsdrivers")
+	operatorClient, dynamicInformers, err := goc.NewClusterScopedOperatorClient(controllerConfig.KubeConfig, gvr)
 	if err != nil {
 		return err
 	}
 
-	ctrlInformers := informers.NewSharedInformerFactoryWithOptions(
-		ctrlClientset,
-		resync,
-		informers.WithTweakListOptions(singleNameListOptions(globalConfigName)),
+	csiControllerSet := csicontrollerset.NewCSIControllerSet(
+		operatorClient,
+		controllerConfig.EventRecorder,
+	).WithLogLevelController().WithManagementStateController(
+		operandName,
+		false,
+	).WithStaticResourcesController(
+		"AWSEBSDriverStaticResources",
+		kubeClient,
+		kubeInformersForNamespaces,
+		generated.Asset,
+		[]string{
+			"namespace.yaml", // Is CSO in charge of creating the namespace?
```

Comment:
Yes, CSO is in charge. It must create the namespace for the operator.

Reply:
Ack